### PR TITLE
Update block design guidelines to include info on setup states

### PR DESCRIPTION
**Pull request by alexislloyd**
Proposed guidelines for when to include a setup state in your block design. Would appreciate feedback on content and validation of markdown syntax.

### CODE REVIEW

File: docs/design/block-design.md
Comment (chrisvanpatten) [inline docs/design/block-design.md]: Only needs one newline around the image :)

Comment (chrisvanpatten) [reply docs/design/block-design.md]: (Applies to all images in the document, fwiw, but I won't note them all as that would be tedious and annoying!)

**Review by chrisvanpatten — COMMENTED** [review]
Small review for markdown syntax. Haven't reviewed content yet but I'm sure it's great 👍

**Comment by jasmussen** [comment]
Overall I dig this a lot, thanks for doing this!

I found myself initially confused by the term "setup state", but after reading through the document, not only was it clear to me, but I also realized that the term we've used so far for this type of content — "placeholder" — may not have been ideal, as _nothing is shown on the frontend when saving placeholder content_. 

Still, I do think it would be worth adding, perhaps just in the introductory paragraph or as a separate paragraph, a small narrative glossary of terms. Could be as simple as "A setup state, also sometimes referred to as _placeholder_", or something in that vein.

Also, final thing — the images used in the gallery, do we know that we have permission to use those? If not, I can find a few, or you can probably find some on https://www.pexels.com/. Note to be careful with Unsplash, I believe their terms are not as open source as we need them to be.

**Comment by jasmussen** [comment]
By the way, this needs a rebase to fix the merge conflict, if you like, I can do that for you tomorrow.

**Comment by jasmussen** [comment]
Rebased!

**Review by jasmussen — APPROVED** [review]
I dig it 👍 👍

I'll let Chris chime in as well. 

Also I think once the checks pass it _should_ be fine to merge, because it's a doc, but CC: @youknowriad just to be sure. Things have changed on the merge front up until very recently.

**Review by youknowriad — APPROVED** [review]

**Review by chrisvanpatten — APPROVED** [review]
This looks great to me!

**Pull request merged**
